Robust.Shared/GameObjects/Systems/SharedMapSystem.Grid.cs

```diff
@@ -909,6 +909,26 @@ public void SetTiles(EntityUid uid, MapGridComponent grid, List<(Vector2i GridIn
         }
     }
 
+    public IEnumerable<TileRef> GetLocalTilesIntersecting(EntityUid uid, MapGridComponent grid, Circle localCircle, bool ignoreEmpty = true,
+        Predicate<TileRef>? predicate = null)
+    {
+        var aabb = new Box2(localCircle.Position.X - localCircle.Radius, localCircle.Position.Y - localCircle.Radius,
+            localCircle.Position.X + localCircle.Radius, localCircle.Position.Y + localCircle.Radius);
+
+        var tileEnumerator = GetLocalTilesEnumerator(uid, grid, aabb, ignoreEmpty, predicate);
+
+        while (tileEnumerator.MoveNext(out var tile))
+        {
+            var tileCenter = tile.GridIndices + grid.TileSizeHalfVector;
+            var direction = tileCenter - localCircle.Position;
+
+            if (direction.LengthSquared() <= localCircle.Radius * localCircle.Radius)
```

Comment:
```suggestion
            if (direction.IsShorterThanOrEqualTo(localCircle.Radius))
```
For improved readability?

Maybe I'm just trying to justify the existence of these helper methods 🙃